### PR TITLE
Make sure the notification text matches the scenario

### PR DESCRIPTION
**Pull request by nickvergessen**
The notification should not advice to change an email, avatar or display name
when the user can not actually change that information.

Fix nextcloud/server#3052

@MorrisJobke quick easy review?

### CODE REVIEW

**Comment by MorrisJobke** [comment]
Code looks good, but I sadly didn't manage to trigger this. Isn't a login and a cron run afterwards enough?

**Comment by nickvergessen** [comment]
It's registers a background job when showing the firstrunwizard.
So afterwards you need to run cron.
Also make sure you don't have displayname, email or avatar, otherwise the notification will be dismissed immediately

File: tests/Notification/NotifierTest.php [hunk 1]
```diff
@@ -183,7 +183,7 @@ public function testPrepare($language, $user, $changeName, $changeAvatar, $name,
 			$notification->expects($this->never())
 				->method('setLink');
 
-			$this->setExpectedException(\InvalidArgumentException::class);
+			$this->expectException(\InvalidArgumentException::class);
```
Comment (MorrisJobke) [inline tests/Notification/NotifierTest.php]:
```
PHP Fatal error:  Call to undefined method OCA\FirstRunWizard\Tests\Notification\NotifierTest::expectException() in /home/travis/build/nextcloud/server/apps/firstrunwizard/tests/Notification/NotifierTest.php on line 186
```

Comment (MorrisJobke) [inline tests/Notification/NotifierTest.php]: This is only broken on PHP 5.6. I would say we should go with the deprecated setExpectedException and then move away from this once 5.6 is not supported anymore (travis ships phpunit 4.5 on PHP 5.6)

Comment (nickvergessen) [inline tests/Notification/NotifierTest.php]: yeah right, I always pick the wrong one. Sorry.

**Pull request merged**